### PR TITLE
Fix unresponsive "Onboarding Questions" screen

### LINKED ISSUE #19774: During the "Log In" flow, the "Onboarding Questions" screen sometimes doesn't respond to user touches
### Expected behavior

"Onboarding Questions" screen should always be responsive.

### Actual behavior

When there is an issue with the user's auth token, the app attempts to resolve the issue by asking the user to log in again, and during that log in flow, the "Onboarding Questions" screen becomes unresponsive and the user gets stuck in that screen.

### Steps to reproduce the behavior

<video src="https://user-images.githubusercontent.com/9609223/207641345-28b7bb53-a4d9-4b8b-838f-10f04af092ea.mp4"></video>

<details><summary><b>Prerequisites</b></summary>
<p>

#### Account Setup
Have an account with at least 2 sites and **2FA enabled**.

#### Siumate Auth Token Issue
1. Log into your account.
2. Go to `WPAuthTokenIssueSolver.m` file and change `hasAuthTokenIssues` to return `true`
```objective-c
- (BOOL)hasAuthTokenIssues
{
//    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
//    WPAccount *account = [WPAccount lookupDefaultWordPressComAccountInContext:context];
//
//    BOOL hasAuthTokenIssues = account && ![account authToken];
//
//    return hasAuthTokenIssues;
    return true;
}
```
3. Build and run the WordPress app
4. **Expect** the Log In screen to appear

</p>
</details>

1.  Follow **Prerequisites** steps 
2. Enter your password
3. **Expect** the 2FA screen to appear
4. Enter 2FA code, then tap Continue
5. **Expect** the Blogs List screen to appear
6. Select any blog
7. **Expect** the “What would you like to focus on first?” screen to appear
8. Nothing happens when the buttons are tapped

### PR DESCRIPTION
**Pull request by salimbraksa**
Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/19774

### Description

This PR resolves an issue that was causing the user to get stuck in the "Onboarding Questions" screen.

| Before | After |
| ------ | ------ |
| <video src="https://user-images.githubusercontent.com/9609223/207641345-28b7bb53-a4d9-4b8b-838f-10f04af092ea.mp4"></video> | <video src="https://user-images.githubusercontent.com/9609223/209358139-5c06c3b0-5bc9-4966-84ee-e5983ad0395b.mp4"></video> |

### Related PR
- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/719

### Test Instructions

<details><summary><b>Prerequisites</b></summary>
<p>

#### Account Setup
Have an account with at least 2 sites and **2FA enabled**.

#### Simulate Auth Token Issue
1. Log into your account.
2. Go to `WPAuthTokenIssueSolver.m` file and change `hasAuthTokenIssues` to return `true`
```objective-c
- (BOOL)hasAuthTokenIssues
{
//    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
//    WPAccount *account = [WPAccount lookupDefaultWordPressComAccountInContext:context];
//
//    BOOL hasAuthTokenIssues = account && ![account authToken];
//
//    return hasAuthTokenIssues;
    return true;
}
```
3. Build and run the WordPress app
4. **Expect** the Log In screen to appear

</p>
</details>

1.  Follow **Prerequisites** steps 
2. Enter your password
3. **Expect** the 2FA screen to appear
4. Enter 2FA code, then tap Continue
5. **Expect** the Blogs List screen to appear
6. Select any blog
7. **Expect** the “What would you like to focus on first?” screen to appear
8. Tap any button
9. **Expect** the next screen to show up

### Regression Notes

1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None.

3. What automated tests I added (or what prevented me from doing so)
None.

### PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by twstokes — APPROVED** [review]
This worked great! I also approved https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/719#pullrequestreview-1234901340. 🚀 

For the pod to get pulled down I had to specify the commit hash rather than the branch. It looks like the branch name is out of sync and still pointing to [98dc0b8](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/commit/98dc0b8bd626d45bf9d174f735564346bef89847) instead of [03e2e89](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/commit/03e2e896f48c0d00416bec4ac5b3150a08b13201) that was force-pushed.

That should get ironed out once the merging / tagging takes place on the WordPressAuthenticator side.

I'd also consider a release note for this. 👍

**Comment by mokagio** [comment]
Since this PR was approved and the auto-merged enabled, the `RELEASE-NOTES.txt` developed conflicts.

I resolved them (with GitHub's web editor, for a change) and am now waiting for the auto-merge to kick in before starting the 21.6 code freeze.

**Pull request merged**
